src/ui/Toggle/ToggleContext.ts

```diff
@@ -0,0 +1,36 @@
+import React, { ChangeEventHandler } from 'react';
+
+/**
+ * The default value of `checked` should be null
+ * to support both case of controlled and uncontrolled component
+ * ref: https://github.com/aaronshaf/react-toggle#props
+ */
+const noop = (): void => { /* noop */ };
+export const TOGGLE_DEFAULT_VALUE = {
+  checked: null,
+  defaultChecked: false,
+  disabled: false,
+  onChange: noop,
+  onFocus: noop,
+  onBlur: noop,
+};
+
+export interface ToggleContextInterface {
+  checked?: boolean;
+  defaultChecked?: boolean;
+  disabled?: boolean;
+  onChange?: ChangeEventHandler<HTMLInputElement>;
+  onFocus?: ChangeEventHandler<HTMLInputElement>;
+  onBlur?: ChangeEventHandler<HTMLInputElement>;
+}
+
+export const ToggleContext = React.createContext<ToggleContextInterface>(TOGGLE_DEFAULT_VALUE);
+
+export function useToggleContext(): ToggleContextInterface {
+  let context = null;
+  try {
+    context = React.useContext(ToggleContext);
+    return context;
+  } catch (_) { }
+  return context;
```

Comment:
Ah.. from previous comment, I wanted to say, its okay to throw error here:
It would be caught at development side while developer is implementing :) 

```
  const context = React.useContext(MessageContext);
  if (context === undefined) {
    throw new Error('useToggleContext must be used within a ToggleContainer');
  }
  return context;
```

Reply:
Ah, yeah I felt I understood wrong. Thanks